baseapp/baseapp.go

```diff
@@ -287,12 +287,19 @@ func (app *BaseApp) storeConsensusParams(consensusParams *abci.ConsensusParams)
 	mainStore.Set(mainConsensusParamsKey, consensusParamsBz)
 }
 
-// getMaximumBlockGas gets the maximum gas from the consensus params.
-func (app *BaseApp) getMaximumBlockGas() (maxGas uint64) {
+// getMaximumBlockGas gets the maximum gas from the consensus params. It panics
+// when the maximum block gas is non-positive.
+func (app *BaseApp) getMaximumBlockGas() uint64 {
 	if app.consensusParams == nil || app.consensusParams.BlockSize == nil {
```

Comment:
this could be shortened, just add `app.consensusParams.BlockSize.MaxGas <= 0`

Reply:
what if `app.consensusParams` or `app.consensusParams.BlockSize` are nil?

Reply:
Well, this is safe:

```golang
if app.consensusParams == nil || app.consensusParams.BlockSize == nil || app.consensusParams.BlockSize.MaxGas <= 0
```